### ISSUE TITLE
Ansible 1.5.2 breaks while gathering facts from FreeNAS 9.2.1.2

### ISSUE THREAD

**Issue by justinjereza** [ansible/ansible#6462]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.5.2
##### Environment:

Running Debian 7.4 and managing FreeNAS 9.2.1.2. Ansible is installed from pypi in a virtualenv.
##### Summary:

Ansible 1.5.2 breaks while gathering facts from FreeNAS 9.2.1.2. All files are in the same directory and ansible-playbook is executed in the same directory. The same files work when using Ansible 1.4.4.
### ansible.cfg

``` ini
[defaults]
hostfile = hosts
remote_user = root
remote_tmp = /tmp/ansible/tmp
```
### hosts

```
san
```
### ping.yml

``` yml

---
- hosts: san
  vars:
    ansible_python_interpreter: /usr/local/bin/python
  tasks:
    - name: test connection
      ping:
```
### ansible-playbook ping.yml

```
PLAY [san] ******************************************************************** 

GATHERING FACTS *************************************************************** 
fatal: [san] => failed to parse: Traceback (most recent call last):
  File "/tmp/ansible/tmp/ansible-tmp-1394688236.61-228120390396989/setup", line 3442, in <module>
    main()
  File "/tmp/ansible/tmp/ansible-tmp-1394688236.61-228120390396989/setup", line 2359, in main
    data = run_setup(module)
  File "/tmp/ansible/tmp/ansible-tmp-1394688236.61-228120390396989/setup", line 2300, in run_setup
    facts = ansible_facts(module)
  File "/tmp/ansible/tmp/ansible-tmp-1394688236.61-228120390396989/setup", line 2291, in ansible_facts
    facts.update(Network(module).populate())
TypeError: __init__() takes exactly 1 argument (2 given)


TASK: [test connection] ******************************************************* 
FATAL: no hosts matched or all hosts have already failed -- aborting


PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/username/ping.retry

san                        : ok=0    changed=0    unreachable=1    failed=0
```

**Comment by HelloGrayson** [comment]
Not entirely positive but I think this might be related to #6430

**Comment by justinjereza** [comment]
It does look like it's related. `facts.update(Network(module).populate())` seems to have appeared in 1.5.1 when it was just `facts.update(Network().populate())` in previous versions.

**Comment by mpdehaan** [comment]
This is already fixed on the 1.5.3 branch and we'll be releasing this today.
